src/lib.rs

```diff
@@ -188,6 +191,8 @@ pub trait Chain<H: Eq, N: Copy + BlockNumberOps> {
 		match self.ancestry(base, block) {
 			Ok(_) => true,
 			Err(Error::NotDescendent) => false,
+			// TODO: probably this whole error handling match should be re-thinked now
+			Err(Error::MultipleForksSupportPredicate) => unreachable!(),
```

Comment:
yup, there's a TODO above about it. Let's just make this function return a result.

Reply:
done, but you probably mean
```rust
Ok(_) => Ok(true),
Err(Error::NotDescendent) => Ok(false),
Err(a) => Err(a),
```